WordPress/Classes/ViewRelated/Me/App Settings/AppSettingsViewController.swift

```diff
@@ -206,6 +205,40 @@ class AppSettingsViewController: UITableViewController {
         }
     }
 
+    func pushImageQualitySettings() -> ImmuTableAction {
+        return { [weak self] row in
+            let values = [MediaSettings.ImageQuality.low,
+                          MediaSettings.ImageQuality.medium,
+                          MediaSettings.ImageQuality.high,
+                          MediaSettings.ImageQuality.maximum]
+
+            let titles = values.map({ (settings: MediaSettings.ImageQuality) -> String in
+                settings.description
+            })
+
+            let currentImageQuality = MediaSettings().imageQualitySetting
+
+            let settingsSelectionConfiguration = [SettingsSelectionDefaultValueKey: currentImageQuality,
+                                                  SettingsSelectionTitleKey: NSLocalizedString("Quality", comment: "The quality of image used when uploading"),
+                                                  SettingsSelectionTitlesKey: titles,
+                                                  SettingsSelectionValuesKey: values] as [String: Any]
+
+            let viewController = SettingsSelectionViewController(dictionary: settingsSelectionConfiguration)
+
+            viewController?.onItemSelected = { (quality: Any!) -> () in
+                let newQuality = quality as! MediaSettings.ImageQuality
+                MediaSettings().imageQualitySetting = newQuality
+
+                // Track setting changes
+                var properties = [String: AnyObject]()
+                properties["quality"] = newQuality.description as String as AnyObject
+                WPAnalytics.track(.appSettingsImageQualityChanged, properties: properties)
```

Comment:
Because [this tracking function](https://github.com/wordpress-mobile/WordPress-iOS/blob/3bc4737bd27414f5509e22a4caf83f601e3272d7/WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift#L1476) takes in `[AnyHashable: Any]` we should be able to do what was done in [`imageOptimizationChanged`](https://github.com/wordpress-mobile/WordPress-iOS/blob/3bc4737bd27414f5509e22a4caf83f601e3272d7/WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift#L1476) after the suggestion is applied. `String` types will "just work" because they conform to `AnyHashable` so no need to cast. 👍

Reply:
Suggestion applied in https://github.com/wordpress-mobile/WordPress-iOS/pull/21981/commits/f29f4d2837d4a1d17b2d4f182f178680d8df74b6.